src/Our.Umbraco.AuthU/Web/Controllers/OAuthController.cs

```diff
@@ -76,6 +77,25 @@ protected void ProcessClient(OAuthTokenRequest request)
                 Client = client;
             }
         }
+	
+	protected void SetOriginHeader()
+	{
+		string allowedOrigin = Client != null ? Client.AllowedOrigin : Context.Options.AllowedOrigin;
+
+		if (HttpContext.Current.Response.Headers.AllKeys.Contains("Access-Control-Allow-Origin"))
+		{
+			var accessControlHeader = HttpContext.Current.Response.Headers.GetValues("Access-Control-Allow-Origin").FirstOrDefault();
+
+			if (!accessControlHeader.Equals(allowedOrigin, StringComparison.OrdinalIgnoreCase))
+			{
+				throw new Exception($"There is currently a header set with the key Access-Control-Allow-Origin, but the value: {accessControlHeader} differs from the OAuth configured value: {allowedOrigin}");
+			}
+		}
+		else
+		{
+			HttpContext.Current.Response.Headers.Add("Access-Control-Allow-Origin", Client != null ? Client.AllowedOrigin : Context.Options.AllowedOrigin);
```

Comment:
Use the `allowedOrigin` variable declared on line 83 instead of doing a null check again.